snapshot/read_write_hdf.py

```diff
@@ -0,0 +1,130 @@
+import h5py
+import numpy as np
+from mpi4py import MPI
+import os
+
+
+class ReadWriteHDF:
+    def __init__(self, logger) -> None:
+        self._logger = logger
+
+    def create_file(self, file_path: str) -> None:
+        """
+        Create a file with a given name in a given directory overriding a file if it exists.
+
+        Parameters
+        ----------
+        dir_name : str
+            Directory in which the file should be created.
+        file_name : str
+            Name of the file to be created.
+
+        """
+        f = h5py.File(file_path, "w")
+        f.close()
+
+    def collect_output_files(self, dir_name: str, file_list: list) -> None:
+        """
+        Iterate over given files in a given directory and copy the content into a single file.
+
+        Parameters
+        ----------
+        dir_name : str
+            Path to directory containing the files.
+        file_list : list
+            List of files to be combined.
+        """
+        # Create a output file
+        main_file = h5py.File(os.path.join(dir_name, "output.hdf5"), "w")
+        # Loop over files
+        for file in file_list:
+            parameter_file = h5py.File(os.path.join(dir_name, file), "r")
+            # Add all data sets to the main file.
+            for key in parameter_file.keys():
+                current_data = parameter_file[key][:]
+                for i in range(len(current_data)):
+                    # If data set does not exist create and add current data
+                    if key not in main_file.keys():
+                        main_file.create_dataset(
+                            key,
+                            data=current_data[i],
+                            shape=(1, *current_data[i].shape),
+                            maxshape=(None, *current_data[i].shape),
+                        )
+                    # If dataset exists resize and add current data
+                    else:
+                        main_file[key].resize(main_file[key].shape[0] + 1, axis=0)
+                        main_file[key][-1] = current_data[i]
+            parameter_file.close()
+        main_file.close()
+
+    def write_sim_output_to_hdf(
+        self, file_path: str, macro_data: dict, micro_data: dict
+    ) -> None:
+        """
+        Write the output of a micro simulation to a hdf5 file.
+
+        Parameters
+        ----------
+        file_path : str
+            Path to file in which the data should be written.
+        macro_data : dict
+            Dict of macro simulation input.
+        micro_data : dict
+            Dict of micro simulation output.
+        """
+        parameter_file = h5py.File(file_path, "a")
+        input_data = [macro_data, micro_data]
```

Comment:
Is this a way to merge two dicts into a new dict? Very nice!

Reply:
No, it creates a list of two dicts. But there are simple ways to merge two dicts into one and it makes sense to do that here as it would allow us to remove one of the for-loops and make the code more readable.

Reply:
Okay. Yes, lets go for merging the two dicts into a new dict and then using that; it will simplify the logic quite a bit.